runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -469,4 +488,64 @@ public void onNpcDespawned(NpcDespawned npcDespawned)
 		NPC npc = npcDespawned.getNpc();
 		npcs.remove(npc);
 	}
+
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired scriptPostFired)
+	{
+		if (scriptPostFired.getScriptId() == ScriptID.AGI_LAPS_CHECK)
```

Comment:
Rather than nesting blocks many levels deep, change each of these successive `if` statements into guard clauses.

ex:
```java
if (scriptPostFired.getScriptId() != ScriptID.AGI_LAPS_CHECK)
{
	return;
}

Widget agiLapsHeader = client.getWidget(WidgetInfo.AGI_LAPS_HEADER);
if (agiLapsHeader == null)
{
	return;
}
// ...
```